roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,354 @@
+"""
+Check that the SDN is routing traffic properly.
+"""
+
+import datetime
+import os
+
+import ipaddress
+from ansible.module_utils import six
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class SDNCheck(NotContainerizedMixin, OpenShiftCheck):
+    """A check to run relevant diagnostics on the SDN."""
+
+    name = 'sdn'
+    tags = ['health']
+
+    def is_active(self):
+        """Skip hosts that are not masters or nodes."""
+        group_names = self.get_var('group_names', default=[])
+        master_or_node = 'oo_masters_to_config' in group_names or \
+                         'oo_nodes_to_config' in group_names
+        return super(SDNCheck, self).is_active() and master_or_node
+
+    def run(self):
+        if self.want_full_results:
+            # Gather diagnostic information and perform diagnostics on a master
+            # or node host.
+            try:
+                self.save_command_output('journal',
+                                         ['/bin/journalctl', '--since=-5m'])
+                self.save_command_output('nmcli-dev',
+                                         ['/bin/nmcli', '--nocheck', '-f', 'all',
+                                          'dev', 'show'])
+                self.save_command_output('nmcli-con',
+                                         ['/bin/nmcli', '--nocheck', '-f', 'all',
+                                          'con', 'show'])
+                self.save_command_output(
+                    'ifcfg',
+                    'head -1000 /etc/sysconfig/network-scripts/ifcfg-*')
+                self.save_command_output('addresses',
+                                         ['/sbin/ip', 'addr', 'show'])
+                self.save_command_output('routes',
+                                         ['/sbin/ip', 'route', 'show'])
+                self.save_command_output('arp',
+                                         ['/sbin/ip', '-s', 'neighbor', 'show'])
+                self.save_command_output('iptables', ['/sbin/iptables-save'])
+                self.register_file('hosts', None, '/etc/hosts')
+                self.register_file('resolv.conf', None, '/etc/resolv.conf')
+                self.save_command_output('modules', ['/sbin/lsmod'])
+                self.save_command_output('sysctl', ['/sbin/sysctl', '-a'])
+                self.save_command_output('docker-version',
+                                         ['/bin/docker', 'version'])
```

Comment:
In case of crio runtime, we need crio version.

Reply:
Added.